src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -1145,11 +1168,16 @@ public void onFailure(Exception e) {
         private void handleEntityResponses() {
             if (failure.get() != null) {
                 listener.onFailure(failure.get());
-            } else if (ackResponses.isEmpty()) {
+            } else if (entityResultResponses.isEmpty()) {
                 listener.onFailure(new InternalFailure(adID, NO_ACK_ERR));
             } else {
                 listener.onResponse(new AnomalyResultResponse(0, 0, 0, new ArrayList<FeatureData>()));
             }
+            long maxTotalUpdates = 0;
+            while (!entityResultResponses.isEmpty()) {
+                maxTotalUpdates = Math.max(maxTotalUpdates, entityResultResponses.poll().getTotalUpdates());
+            }
+            adTaskManager.updateLatestRealtimeTask(adID, maxTotalUpdates, intervalMinutes, failure.get());
```

Comment:
similar to the other comments, should we leave the setting error logic to job scheduler? This error can be overwritten by other errors (e.g., feature aggregation and other pages's entityResultResponse).

Reply:
Same as the comment above